Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,317 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Text.Json;
+
+/// <summary>
+/// Provides extension methods for converting API Gateway responses to HttpResponse objects.
+/// </summary>
+public static class ApiGatewayResponseExtensions
+{
+
+    private const string InternalServerErrorMessage = "{\"message\":\"Internal Server Error\"}";
+
+    /// <summary>
+    /// Converts an APIGatewayProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway proxy response to convert.</param>
+    /// <returns>An HttpResponse representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayProxyResponse apiResponse, ApiGatewayEmulatorMode emulatorMode)
+    {
+        var httpContext = new DefaultHttpContext();
+        var response = httpContext.Response;
+
+        SetResponseHeaders(response, apiResponse.Headers, emulatorMode, apiResponse.MultiValueHeaders);
+        SetResponseBody(response, apiResponse.Body, apiResponse.IsBase64Encoded, emulatorMode);
+        SetContentTypeAndStatusCode(response, apiResponse.Headers, apiResponse.MultiValueHeaders, apiResponse.StatusCode, emulatorMode);
+
+        return response;
+    }
+
+    /// <summary>
+    /// Converts an APIGatewayHttpApiV2ProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway HTTP API v2 proxy response to convert.</param>
+    /// <returns>An HttpResponse representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayHttpApiV2ProxyResponse apiResponse)
+    {
+        var httpContext = new DefaultHttpContext();
+        var response = httpContext.Response;
+
+        SetResponseHeaders(response, apiResponse.Headers, ApiGatewayEmulatorMode.HttpV2);
+        SetResponseBody(response, apiResponse.Body, apiResponse.IsBase64Encoded, ApiGatewayEmulatorMode.HttpV2);
+        SetContentTypeAndStatusCodeV2(response, apiResponse.Headers, apiResponse.Body, apiResponse.StatusCode);
+
+        return response;
+    }
+    /// <summary>
+    /// Sets the response headers on the HttpResponse, including default API Gateway headers based on the emulator mode.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set headers on.</param>
+    /// <param name="headers">The single-value headers to set.</param>
+    /// <param name="multiValueHeaders">The multi-value headers to set.</param>
+    /// <param name="emulatorMode">The API Gateway emulator mode determining which default headers to include.</param>
+    private static void SetResponseHeaders(HttpResponse response, IDictionary<string, string>? headers, ApiGatewayEmulatorMode emulatorMode, IDictionary<string, IList<string>>? multiValueHeaders = null)
+    {
+        var processedHeaders = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
If you;re using the HttpResponse from the API, you might need to clear the existing headers first